common/formatting/align_test.cc

```diff
@@ -975,5 +976,450 @@ TEST_F(InferAmbiguousAlignIntentTest, DifferenceSufficientlySmall) {
+ public:
+  SyntaxTreeColumnizer() = default;
+
+  void Visit(const SyntaxTreeNode& node) override {
+    ColumnPositionTree* column;
+    if (!current_column_)
+      column = ReserveNewColumn(node, props);
+    else
+      column = ReserveNewColumn(current_column_, node, props);
+
+    ValueSaver<ColumnPositionTree*> current_column_saver(&current_column_,
+                                                         column);
+    ColumnSchemaScanner::Visit(node);
+  }
+  void Visit(const SyntaxTreeLeaf& leaf) override {
+    AlignmentColumnProperties local_props = props;
+    if (leaf.get().text() == ",") local_props.contains_delimiter = true;
+
+    if (!current_column_)
+      ReserveNewColumn(leaf, local_props);
+    else
+      ReserveNewColumn(current_column_, leaf, local_props);
+  }
+
+ private:
+  ColumnPositionTree* current_column_ = nullptr;
+};
+
+class SubcolumnsTreeAlignmentTest : public MatrixTreeAlignmentTestFixture {
+ public:
+  SubcolumnsTreeAlignmentTest(absl::string_view text =
+                                  "zero\n"
+                                  "( one two three )\n"
+                                  "( four ( five six ) seven )\n"
+                                  "( eight ( ( nine ) ten ) )\n"
+                                  "( eleven nineteen-ninety-nine 2k )\n")
+      : MatrixTreeAlignmentTestFixture(text) {
+    //  Columns tree:
+    //
+    //  ┃┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┃
+    //  ┃┄┃┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┃┄┃
+    //  ┊ ┃┄┄┄┄┄┄┃┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┃┄┄┄┄┄┃ ┊
+    //  ┊ ┊      ┃┄┃┄┄┄┄┄┄┄┄┄┄┄┄┃┄┃   ┊     ┊ ┊
+    //  ┊ ┊      ┊ ┃┄┄┄┄┄┄┄┄┃┄┄┄┃ ┊   ┊     ┊ ┊
+    //  ┊ ┊      ┊ ┃┄┃┄┄┄┄┃┄┃   ┊ ┊   ┊     ┊ ┊
+    //  ┊ ┊      ┊ ┊ ┃┄┄┄┄┃ ┊   ┊ ┊   ┊     ┊ ┊
+    //  ┃zero    ┊ ┊ ┊    ┊ ┊   ┊ ┊   ┊     ┊ ┃
+    //  ┃(┃one   ┃two┊    ┊ ┊   ┊ ┊   ┃three┃)┃
+    //  ┃(┃four  ┃(┃five  ┊ ┃six┃)┃   ┃seven┃)┃
+    //  ┃(┃eight ┃(┃(┃nine┃)┃ten┃)┃   ┃     ┃)┃
+    //  ┃(┃eleven┃nineteen-ninety-nine┃2k   ┃)┃
+    //  ╵ ╵      ╵ ╵ ╵    ╵ ╵   ╵ ╵   ╵     ╵ ╵
+
+    syntax_tree_ = TNode(0);
+
+    UnwrappedLine all(0, pre_format_tokens_.begin());
+    all.SpanUpToToken(pre_format_tokens_.end());
+    all.SetOrigin(syntax_tree_.get());
+    partition_ = TokenPartitionTree{all};
+
+    auto token_iter = pre_format_tokens_.begin();
+    while (true) {
+      UnwrappedLine uwline(0, token_iter);
+      SymbolPtr item = ParseItem(&token_iter, pre_format_tokens_.end());
+      if (!item.get()) {
+        break;
+      }
+      uwline.SpanUpToToken(token_iter);
+      uwline.SetOrigin(item.get());
+      partition_.NewChild(uwline);
+      SymbolCastToNode(*syntax_tree_).AppendChild(std::move(item));
+    }
+  }
+
+ private:
+  SymbolPtr ParseList(
+      std::vector<verible::PreFormatToken>::iterator* it,
+      const std::vector<verible::PreFormatToken>::iterator& end) {
+    SymbolPtr list = TNode(0);
+    SymbolPtr item;
+    while ((item = ParseItem(it, end)).get() != nullptr) {
```

Comment:
Isn't tha `get()` call somewhat redundant, as the smart-pointer already has an overload for comparing against `nullptr` ?

Reply:
It is. Removed this one and the one a few lines above (also null check)